kasa/smart/smartdevice.py

```diff
@@ -645,3 +661,172 @@ async def set_fan_speed_level(self, level: int):
+
+    @property
+    def color_temp(self) -> int:
+        """Whether the bulb supports color temperature changes."""
+        if not self.is_variable_color_temp:
+            raise KasaException("Bulb does not support colortemp.")
+
+        return cast(
+            ColorTemperatureModule, self.modules["ColorTemperatureModule"]
+        ).color_temp
+
+    @property
+    def brightness(self) -> int:
+        """Return the current brightness in percentage."""
+        if not self.is_dimmable:  # pragma: no cover
+            raise KasaException("Bulb is not dimmable.")
+
+        return self._info.get("brightness", -1)
+
+    async def set_hsv(
+        self,
+        hue: int,
+        saturation: int,
+        value: int | None = None,
+        *,
+        transition: int | None = None,
+    ) -> dict:
+        """Set new HSV.
+
+        Note, transition is not supported and will be ignored.
+
+        :param int hue: hue in degrees
+        :param int saturation: saturation in percentage [0,100]
+        :param int value: value between 1 and 100
+        :param int transition: transition in milliseconds.
+        """
+        if not self.is_color:
+            raise KasaException("Bulb does not support color.")
+
+        return await cast(ColorModule, self.modules["ColorModule"]).set_hsv(
+            hue, saturation, value
+        )
+
+    async def set_color_temp(
+        self, temp: int, *, brightness=None, transition: int | None = None
+    ) -> dict:
+        """Set the color temperature of the device in kelvin.
+
+        Note, transition is not supported and will be ignored.
+
+        :param int temp: The new color temperature, in Kelvin
+        :param int transition: transition in milliseconds.
+        """
+        if not self.is_variable_color_temp:
+            raise KasaException("Bulb does not support colortemp.")
+        return await cast(
+            ColorTemperatureModule, self.modules["ColorTemperatureModule"]
+        ).set_color_temp(temp)
+
+    async def set_brightness(
+        self, brightness: int, *, transition: int | None = None
+    ) -> dict:
+        """Set the brightness in percentage.
+
+        Note, transition is not supported and will be ignored.
+
+        :param int brightness: brightness in percent
+        :param int transition: transition in milliseconds.
+        """
+        if not self.is_dimmable:  # pragma: no cover
+            raise KasaException("Bulb is not dimmable.")
+
+        return await cast(Brightness, self.modules["Brightness"]).set_brightness(
+            brightness
+        )
+
+    async def set_effect(
+        self,
+        effect: str,
+        *,
+        brightness: int | None = None,
+        transition: int | None = None,
+    ) -> None:
+        """Set an effect on the device."""
+        raise NotImplementedError()
+
+    @property
+    def presets(self) -> list[BulbPreset]:
+        """Return a list of available bulb setting presets."""
+        return []
```

Comment:
We should remove this, too. We can add something similar back using features, if that even makes much sense (as those were not really used for anything, afaik).

Reply:
This is implemented on the `IotBulb` and is in the main interface as well as the documentation for the `IotBulb` which we probably want to move to `Bulb`.  Maybe we should keep it for now otherwise we need to remove it from the base interface and I'm sure we can get it working shortly,

Reply:
Okay, let's keep it for the time being :+1: